packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -19,28 +19,26 @@ export const makeZoeHelpers = zoe => {
 
   // Compare the keys of actual with expected keys and reject offer if
   // not sameStructure. If expectedKeys is undefined, no comparison occurs.
-  const rejectIf = (
+  const rejectKeysIf = (
     inviteHandle,
     actual,
-    expectedKeys,
+    expected,
     msg = defaultRejectMsg,
     // eslint-disable-next-line consistent-return
   ) => {
-    if (expectedKeys !== undefined) {
-      const expected = [...expectedKeys]; // in case hardened
-      expected.sort();
-      if (!sameStructure(getKeysSorted(actual), harden(expected))) {
+    if (expected !== undefined) {
+      if (!sameStructure(getKeysSorted(actual), getKeysSorted(expected))) {
         return rejectOffer(inviteHandle, msg);
       }
     }
   };
-  // Compare actual to expected keys. If expectedKeys is
+  // Compare actual keys to expected keys. If expectedKeys is
   // undefined, return true trivially.
-  const check = (actual, expectedKeys) => {
-    if (expectedKeys === undefined) {
+  const checkKeys = (actual, expected) => {
+    if (expected === undefined) {
       return true;
     }
-    return sameStructure(getKeys(actual), expectedKeys);
+    return sameStructure(getKeys(actual), getKeys(expected));
```

Comment:
Does this need to be sorted too? I realize that's not changed here, but maybe we should fix it.

Reply:
Maybe we could add a test that sees what happens if getKeys(actual) is in a different order than getKeys(expected)?

Reply:
`getKeys()` extracts an (ordered) array from an (unordered) record. `sameStructure()` detects order differences. I added a test.

Reply:
Ok, you mean `getKeysSorted`, right? I don't think `getKeys` is ordered by anything other than insertion order. Thanks for adding the test!

Reply:
`getKeys()` was what I used at first, and what you responded to. When I wrote
> getKeys() extracts an (ordered) array from an (unordered) record. sameStructure() detects order differences. I added a test.

I was explaining the behavior of `getKeys()`. It takes a record (which is unordered), and produces an array (which is ordered). Since `sameStructure()` is order-sensitive, that implicit conversion has the problem you identified. So after adding a test, I changed the method to use `getKeysSorted()`, as you expected.

Reply:
Got it, by ordered, I thought you meant sorted